app/Http/ViewHelpers/Adminland/AdminSoftwareViewHelper.php

```diff
@@ -0,0 +1,166 @@
+            'product_key' => $software->product_key,
+            'seats' => $software->seats,
+            'licensed_to_name' => $software->licensed_to_name,
+            'licensed_to_email_address' => $software->licensed_to_email_address,
+            'purchase_amount' => $software->purchase_amount / 100,
+            'currency' => $software->currency,
+            'website' => $software->website,
+            'purchased_date_year' => $software->purchased_at ? $software->purchased_at->year : null,
+            'purchased_date_month' => $software->purchased_at ? $software->purchased_at->month : null,
+            'purchased_date_day' => $software->purchased_at ? $software->purchased_at->day : null,
+        ];
+    }
+
+    /**
+     * Get the employees who are assigned with this software.
+     *
+     * @param mixed $employees
+     * @param Company $company
+     * @return Collection
+     */
+    public static function seats($employees, Company $company): Collection
+    {
+        $employeesCollection = collect();
+        foreach ($employees as $employee) {
+            $employeesCollection->push([
+                'id' => $employee->id,
+                'name' => $employee->name,
+                'avatar' => ImageHelper::getAvatar($employee, 21),
+                'product_key' => $employee->pivot->product_key,
+                'url' => route('employees.show', [
+                    'company' => $company,
+                    'employee' => $employee,
+                ]),
+            ]);
+        }
+
+        return $employeesCollection;
+    }
+
+    /**
+     * Get the list of potential employees who can be assigned the software.
+     *
+     * @param Software $software
+     * @param Company $company
+     * @param string $criteria
+     * @return Collection
+     */
+    public static function getPotentialEmployees(Software $software, Company $company, string $criteria): Collection
+    {
+        // get list of employees who have the software
+        $employees = $software->employees()
+            ->select('id')
+            ->pluck('id')
+            ->toArray();
+
+        $potentialEmployees = $company->employees()
+            ->select('id', 'first_name', 'last_name', 'avatar_file_id')
+            ->notLocked()
+            ->whereNotIn('id', $employees)
+            ->where(function ($query) use ($criteria) {
+                $query->where('first_name', 'LIKE', '%' . $criteria . '%')
+                    ->orWhere('last_name', 'LIKE', '%' . $criteria . '%')
+                    ->orWhere('email', 'LIKE', '%' . $criteria . '%');
+            })
+            ->orderBy('last_name', 'asc')
+            ->take(10)
+            ->get();
+
+        $employeesCollection = collect([]);
+        foreach ($potentialEmployees as $employee) {
+            $employeesCollection->push([
+                'id' => $employee->id,
+                'name' => $employee->name,
+            ]);
+        }
+
+        return $employeesCollection;
```

Comment:
Use laravel's collection map function
```suggestion
        return $potentialEmployees->map(function ($employee) {
            return [
                'id' => $employee->id,
                'name' => $employee->name,
            ];
        });
```